src/status_im2/contexts/wallet/events.cljs

```diff
@@ -0,0 +1,40 @@
+(ns status-im2.contexts.wallet.events
+  (:require [utils.re-frame :as rf]
+            [taoensso.timbre :as log]
+            [utils.security.core :as security]
+            [native-module.core :as native-module]))
+
+(rf/reg-event-fx 
+:wallet-2/create-derived-addresses
+                 (fn [{:keys [db]} [password {:keys [path]} on-success]]
+                   (let [{:keys [wallet-root-address]} (:profile/profile db)
+                         sha3-pwd   (native-module/sha3 (str (security/safe-unmask-data password)))]
+                     {:json-rpc/call [{:method     "wallet_getDerivedAddresses"
+                                       :params     [sha3-pwd (clj->js wallet-root-address) (clj->js [path])]
+                                       :on-success on-success
+                                       :on-error   #(log/info "failed to derive address " %)}]})))
+(rf/reg-event-fx :wallet-2/add-account
+                 (fn [{:keys [db]} [password {:keys [emoji account-name color]} {:keys [public-key address  path]}]]
+                   (let [key-uid   (get-in db [:profile/profile :key-uid])
+                         sha3-pwd   (native-module/sha3  (security/safe-unmask-data password))
+                         account-config (clj->js {:key-uid key-uid
+                                                  :wallet false
+                                                  :chat false
+                                                  :type :generated
+                                                  :name account-name
+                                                  :emoji emoji
+                                                  :path path
+                                                  :address  address
+                                                  :public-key public-key
+                                                  :colorID color})]
+                     {:json-rpc/call [{:method     "accounts_addAccount"
+                                       :params     [sha3-pwd account-config]
+                                       :on-success #(rf/dispatch [:navigate-to :wallet-accounts])
+                                       :on-error   #(log/info "failed to create account " %)}]})))
+
+(rf/reg-event-fx
+ :wallet-2/derive-address-and-add-account
+ (fn [_ [password  account-details]]
+   (let [on-success  (fn [derived-adress-details]
+                       (rf/dispatch [:wallet-2/add-account password account-details (first derived-adress-details)]))]
+     (rf/dispatch [:wallet-2/create-derived-addresses password account-details on-success]))))
```

Comment:
@J-Son89, in re-frame, event handlers should never directly call `re-frame.core/dispatch`, only effects should do that (`:json-rpc/call` is one example that does this behind the scenes). The event handler should be pure and always either return nil or a map of effects.

I updated the guidelines recently:

> Register events with `re-frame.core/reg-event-fx` and follow [re-frame's best
practice](https://github.com/day8/re-frame/blob/39adca93673f334dc751ee2d99d340b51a9cc6db/docs/FAQs/BestPractice.md#use-the-fx-effect) so use only `:db` and `:fx` effects.
>
> -- https://github.com/status-im/status-mobile/blob/6e897f0ea6113f8242d4180f52eae6711a9c9938/doc/new-guidelines.md#registering-event-handlers

So here the code would be:

```clojure
(rf/reg-event-fx :wallet-2/derive-address-and-add-account
 (fn [_ [password account-details]]
   (let [on-success (fn [derived-adress-details]
                      (rf/dispatch [:wallet-2/add-account password account-details
                                    (first derived-adress-details)]))]
     {:fx [[:dispatch [:wallet-2/create-derived-addresses password account-details on-success]]]})))
```